code/obj/flock/cage.dm

```diff
@@ -27,19 +26,38 @@
 	mat_changeappearance = 0
 
 
-	New(loc, mob/living/iced as mob, datum/flock/F=null)
-		..()
-		src.flock = F
+	New(loc, var/atom/iced, datum/flock/F=null)
+		..(loc,F)
+		if(iced && !isAI(iced) && !isblob(iced) && !iswraith(iced))
+			if(istype(iced.loc, /obj/flock_structure/cage)) //Already in a cube?
+				qdel(src)
+				return
+
+			if(!(ismob(iced) || iscritter(iced)))
+				qdel(src)
+				return
+			iced:set_loc(src)
+
+			boutput(iced, "<span class='alert'>You are trapped within [src]!</span>") // since this is used in at least two places to trap people in things other than ice cubes
+
 		var/datum/reagents/R = new /datum/reagents(initial_volume)
 		src.reagents = R
 		R.my_atom = src //grumble
 		if(iced)
-			iced.addOverlayComposition(/datum/overlayComposition/flockmindcircuit)
+			if(istype(iced,/mob/living))
+				var/mob/living/M = iced
+				M.addOverlayComposition(/datum/overlayComposition/flockmindcircuit)
 			occupant = iced
-		processing_items |= src
-		src.setMaterial(getMaterial("gnesis"))
-		src.health = src.health_max
-		src.AddComponent(/datum/component/flock_protection, FALSE, TRUE, TRUE)
+
+		UpdateIcon()
+
+	update_icon()
+		. = ..()
+		src.underlays = list()
+		for(var/atom/O in src.contents)
+			src.underlays += O.appearance
+		UpdateOverlays(image(src.icon,src.icon_state,layer = EFFECTS_LAYER_UNDER_4),"icecube_layer")
+
 
 	proc/getHumanPiece(var/mob/living/carbon/human/H)
```

Comment:
Some more list.len >= 1 -> length(list) and list.len -> lenth(list) that can changed in this proc